lib/dynamoid/criteria/chain.rb

```diff
@@ -311,20 +311,17 @@ def key_present?
       # If using a secondary index then we must include the index's composite key
       # as well as the tables composite key.
       def start_key
+        return @start if @start.is_a?(Hash)
         hash_key = @hash_key || source.hash_key
         range_key = @range_key || source.range_key
 
         key = {}
-        key[:hash_key_element] = type_cast_condition_parameter(hash_key, @start.send(hash_key))
-        key[:range_key_element] = type_cast_condition_parameter(range_key, @start.send(range_key)) if range_key
+        key[hash_key] = type_cast_condition_parameter(hash_key, @start.send(hash_key))
+        key[range_key] = type_cast_condition_parameter(range_key, @start.send(range_key)) if range_key
+        # Add table composite keys if they differ from secondary index used composite key
+        key[source.hash_key] = type_cast_condition_parameter(source.hash_key, @start.hash_key) if hash_key != source.hash_key
 
```

Comment:
Can probably remove this blank line if you do one liners, kind of prefer we have lines less than 120 characters though, which is why original line had it split into multi-line if statement.